crates/ntx-builder/src/state/mod.rs

```diff
@@ -116,13 +116,14 @@ impl State {
 
         let notes = state.store.get_unconsumed_network_notes().await?;
         for note in notes {
-            let prefix = note.account_prefix();
-
-            // Ignore notes which don't target an existing account.
-            let Some(account) = state.fetch_account(prefix).await? else {
-                continue;
-            };
-            account.add_note(note);
+            // TODO: support multi target network notes.
+            if let Some(prefix) = note.account_prefix() {
```

Comment:
This does feel a bit weird because its effectively an enum match but we're pushing the match inside the function. I would prefer having a strong type for this instead. This would also let us be confident in the other code areas where right now we still take a `NetworkNote` and just assume its single target from external usage.